File: src/model/History.java
package model;

import java.time.Instant;

public class History {
    private Instant timestamp;
    private OperationType operation;
    private Wallet wallet;


    public History(OperationType operation, Wallet wallet) {
        this.timestamp = Instant.now();
        this.operation = operation;
        this.wallet = wallet;
    }

    public Instant getTimestamp() {
        return timestamp;
    }

    public void setTimestamp(Instant timestamp) {
        this.timestamp = timestamp;
    }

    public OperationType getOperation() {
        return operation;
    }

    public void setOperation(OperationType operation) {
        this.operation = operation;
    }

    public Wallet getWallet() {
        return wallet;
    }

    public void setWallet(Wallet wallet) {
        this.wallet = wallet;
    }
}

File: src/model/history.py
from datetime import datetime

from src.model.operation_type import OperationType
from src.model.wallet import Wallet


class History:
    def __init__(self, operation: OperationType, wallet: Wallet):
        self.__timestamp: datetime = datetime.now()
        self.__operation: OperationType = operation
        self.__wallet: Wallet = wallet

    @property
    def timestamp(self) -> datetime:
        return self.__timestamp

    @property
    def operation(self) -> OperationType:
        return self.__operation

    @operation.setter
    def operation(self, operation: OperationType) -> None:
        self.__operation = operation

    @property
    def wallet(self) -> Wallet:
        return self.__wallet

    @wallet.setter
    def wallet(self, wallet: Wallet) -> None:
        self.__wallet = wallet
